### PR TITLE
Added mini-calender

### PR DESCRIPTION
**Pull request by mly32**
### Summary 

Created mini-calendar to set the current date. The set date **IS** linked to the Export, Scheduler, and Unscheduled Rides views.

### Test Plan 
image

Try setting dates and viewing on different screen sizes. Test Export, Scheduler, and Unscheduled Rides views to make sure everything is displaying correctly.

### Notes 

To force the modal to close, the code uses a scroll. The button widths are constants.

Used context to share date information. I am unsure if there will be performance issues due to the way I wrote it.

### Other Fixes

Fixed scheduler view to show all events on full view and some useEffect issues.

### Breaking Changes  

None

### CODE REVIEW

**Review by mg2239 — CHANGES_REQUESTED** [review]
code looks good! just a couple style things:
image

- green text for today/tomorrow
- use cal icon provided in design
- match font size in design

image
 - circle indicators feasible?
 - match button padding and font size w/ design

also use rem whenever possible, as Pi has an approved PR that changes all the px to rem

**Comment by mly32** [comment]
> code looks good! just a couple style things:
> image
> 
> * green text for today/tomorrow
> * use cal icon provided in design
> * match font size in design
> 
> image
> 
> * circle indicators feasible?
> * match button width, top + bottom padding, and font size w/ design
> 
> also use rem whenever possible, as Pi has an approved PR that changes all the px to rem

File: frontend/src/components/MiniCal/MiniCal.tsx [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+          <span className={styles.primary}>
+            {isToday(curDate) ? 'Today ' : ' '}
+            {isTomorrow(curDate) ? 'Tomorrow ' : ' '}
+          </span>
+          <span className={styles.space} /> <Icon />
+          <span className={styles.space} /> {this.props.value}
+        </button>
+      );
+    }
+  }
```
Comment (mg2239) [inline frontend/src/components/MiniCal/MiniCal.tsx:50]: any reason why this is a class component vs functional component?

File: frontend/src/components/MiniCal/MiniCal.tsx [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+      fill="white"
+    />
+  </svg>
```
Comment (mg2239) [inline frontend/src/components/MiniCal/MiniCal.tsx:28]: put this in .svg file and use w/ img tag

**Review by mg2239 — APPROVED** [review]
made some style changes, just a couple other things and should be good to go

**Pull request merged**